p2p-transfer/src/node.rs

```diff
@@ -103,19 +126,74 @@ impl Echo {
 
     }
 
-    async fn handle_connection_0(&self, connection: &Connection) -> Result<()> {
+    async fn handle_connection_0(&self, connection: &Connection) -> std::result::Result<(), AcceptError> {
 
         let node_id = connection.remote_node_id()?;
         info!("Accepted connection from {}", node_id);
 
         let (mut send, mut recv) = connection.accept_bi().await?;
 
-        //Echo any bytes received
-        let bytes_sent = tokio::io::copy(&mut recv, &mut send).await?;
+        // Read filename length
+        let mut name_len_buf = [0u8; 4];
+        recv.read_exact(&mut name_len_buf).await.map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
+        let name_len = u32::from_le_bytes(name_len_buf) as usize;
+
+        // Read filename
+        let mut name_buf = vec![0u8; name_len];
+        recv.read_exact(&mut name_buf).await.map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
+        let _received_file_name = String::from_utf8(name_buf).map_err(|e| std::io::Error::new(std::io::ErrorKind::InvalidData, e))?;
+
+        // Read file data length
+        let mut data_len_buf = [0u8; 8];
+        recv.read_exact(&mut data_len_buf).await.map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
+        let data_len = u64::from_le_bytes(data_len_buf) as usize;
+
+        // Read file data
+        let mut _received_file_data = vec![0u8; data_len];
```

Comment:
send and receive in chunks so we don't keep so much in ram and as we receive save to disk se we don' tkeep the content of all files in ram